app/models/user.rb

```diff
@@ -1,6 +1,23 @@
 class User < ActiveRecord::Base
 
-  def fullname
-    first_name + " " + last_name
+  has_many :massages,    inverse_of: :user
+  has_many :treatments,  inverse_of: :user
+  has_one  :facial_hair, inverse_of: :user
+
+  validates :first_name, :last_name, :email, presence: true
+  validates_uniqueness_of :email
+
+  def self.search(needle)
+    needle = "%#{needle}%"
+    users  = Arel::Table.new(:users)
+    query  = users[:email].matches(needle)
+         .or(users[:first_name].matches(needle)
+         .or(users[:last_name].matches(needle)))
+    where(query).order(users[:updated_at].desc)
   end
+
+  def facial_hair(*args)
```

Comment:
I don't recommend this approach.  This could lead to innocuous bugs later down the line.
```
user.facial_hair.present?
   ... do something ...
else
   ... do something when they don't have this created ...
end
```
This is a common developer booby trap.  Overriding the AR default behavior with something a bit magical.  This looks a bit innocuous now, but later on you will forget, or some other developer will be scratching their head about some weird bug as a result of using this method.
I'd suggest ```find_or_build_facial_hair(*args)``` as a clearer name.